core/frontend/apps/private-blogging/lib/middleware.js

```diff
@@ -147,6 +155,24 @@ const privateBlogging = {
             };
             return next();
         }
+    },
+
+    /**
+     * We should never render a 404 error for private sites, as these can leak information
+     */
+    handle404: function handle404(err, req, res, next) {
```

Comment:
This handler breaks 404 pages when authenticated to the private site:
![image](https://user-images.githubusercontent.com/3798302/84758005-c90ce780-afbc-11ea-9107-3d5a974562b8.png)

Reply:
Good catch Matt! Had totally missed this while trying to focus on breaking in as unauthenticated 🙈